### DISCUSSION TITLE
bump Zabbix

### DISCUSSION THREAD

**Discussion by notlebowski** [Freetz-NG/freetz-ng#761] [Unerfüllte Wünsche] [answered]
I created a new (virtual) machine with Ubuntu 22 because I want to start using Freetz-NG instead of the original one, so I can compile modern Freetz firmwares.
The whole purpose of running Freetz is running a Zabbix agent which can be used to monitor the networks of the ~60 Fritzboxes I have in the field.

First I want to undo all the changes, by deleting .config

```
cd freetz-ng
rm .config
make menuconfig
```

I then change 2 options  "Level of user competence" to "Expert" and "Target region" to "(avme - international)" and leave "menuconfig".

I then copy ".config" to ".config.default"

```
cp -p .config .config.default
make

```
It then creates me a 7590 image

```
packing images/7590_07.50.all_freetz-ng-21363-d78ccc24b_20230308-064040.image
  packed image file size: 37.6 MB (39434240 bytes)
signing packed .image file
  signed image file size: 37.6 MB (39434240 bytes)
source firmware: 7590_de-es-fr-it-nl-pl 154.07.50 rev101716 {ALL} [MOVE21] (21.11.2022 08:21:05)
  source image file size: 37.0 MB (38840320 bytes)
done.

FINISHED

```
All good up to here.
Before I enabled all kind of packages and the compile wasn't successful. Now I only want to enable the "Zabbix package" as that's essential to me.

```
diff .config .config.default
639,644c639
< FREETZ_PACKAGE_ZABBIX=y
< FREETZ_PACKAGE_ZABBIX_agentd=y
< # FREETZ_PACKAGE_ZABBIX_server is not set
< # FREETZ_PACKAGE_ZABBIX_proxy is not set
< # FREETZ_PACKAGE_ZABBIX_sender is not set
< # FREETZ_PACKAGE_ZABBIX_get is not set
---
> # FREETZ_PACKAGE_ZABBIX is not set

make

```

The compile does not succeed giving this error message

```
_SOURCE -D_FILE_OFFSET_BITS=64 -Wl,-I/usr/lib/freetz/ld-uClibc.so.1     -I/home/freetz/freetz-ng/toolchain/build/mips_gcc-8.3.0_uClibc-1.0.42-nptl_kernel-4.9/mips-linux-uclibc/usr/include          -ffunction-sections -fdata-sections -MT libspecsysinfo_a-diskspace.o -MD -MP -MF .deps/libspecsysinfo_a-diskspace.Tpo -c -o libspecsysinfo_a-diskspace.o `test -f 'diskspace.c' || echo './'`diskspace.c
cpu.c:68:12: error: static declaration of 'getloadavg' follows non-static declaration
 static int getloadavg (double loadavg[], int nelem)
            ^~~~~~~~~~
In file included from ../../../../include/sysinc.h:30,
                 from ../../../../include/common.h:23,
                 from cpu.c:20:
/home/freetz/freetz-ng/toolchain/build/mips_gcc-8.3.0_uClibc-1.0.42-nptl_kernel-4.9/mips-linux-uclibc/include/stdlib.h:915:12: note: previous declaration of 'getloadavg' was here
 extern int getloadavg (double __loadavg[], int __nelem)
            ^~~~~~~~~~
make[6]: *** [Makefile:482: libspecsysinfo_a-cpu.o] Error 1
make[6]: *** Waiting for unfinished jobs....
mv -f .deps/libspecsysinfo_a-diskspace.Tpo .deps/libspecsysinfo_a-diskspace.Po
make[5]: *** [Makefile:590: all-recursive] Error 1
make[4]: *** [Makefile:456: all-recursive] Error 1
make[3]: *** [Makefile:432: all-recursive] Error 1
make[2]: *** [Makefile:444: all-recursive] Error 1
make[1]: *** [make/pkgs/zabbix/zabbix.mk:43: source/target-mips_gcc-8.3.0_uClibc-1.0.42-nptl_kernel-4.9/zabbix-2.4.6/src/zabbix_agent/zabbix_agentd] Terminated
make: *** [Makefile:46: envira] Terminated

```

I hope it gets resolved, so I can switch to Freetz-NG.
I've used Freetz for +11 years on many boxes and these are currently operational.

**Comment by fda77** [comment]
this package has no maintainer since 8 years. if you want to use it you should use a current version, which should compile with recent compiler, libc and libs

**Comment by notlebowski** [comment, marked as answer]
I'm not a developer. 
If you have no interest in making it work for Freetz-NG. I have no interest in Freetz-NG...  or should I say Freetz lost generation?

I thought it was supposed to be better?
I already suspected something fishy when I noticed there aren't any open tickets.

This is not a fork....  it's a sword which decapitates a project.

**Comment by fda77** [comment]
I've 0 interest to do anything for you. Or in short: fuck you
